feat: Add Files API support for file upload and management

cxyfer opened the pull request:
## 🚀 Feature Description

Implements Google Gemini [Files API](https://ai.google.dev/api/files) support with `v1beta/files` endpoints, enabling file upload, list, get, and delete operations. This implementation addresses the need for large file uploads, particularly for audio transcription use cases.

Closes #198

## 🔧 Key Changes

- **New endpoints**: 
  - `POST /upload/v1beta/files` - Initialize and handle file uploads
  - `GET /v1beta/files` - List uploaded files
  - `GET /v1beta/files/{file_id}` - Get file metadata
  - `DELETE /v1beta/files/{file_id}` - Delete a file
- **File service**: Upload handling with resumable transfer support for large files
- **Database integration**: `FileRecord` model with state tracking (PROCESSING, ACTIVE, FAILED)
- **User isolation**: Files are tracked per `user_token` to ensure data isolation
- **Auto cleanup**: Scheduled task for expired file removal
- **Fixed API key**: Uses dedicated key for Files API due to Google restrictions

## ✅ Testing

- [x] Upload test: POST 20MB video file, verify resumable upload
- [x] List test: GET files with pagination
- [x] Get test: Retrieve individual file metadata
- [x] Delete test: Remove file and verify deletion
- [x] Scheduled task: Verify automatic cleanup of expired files

Test script: https://gist.github.com/cxyfer/c108e84030bb5d740eb517e52fc6b7d8

cxyfer committed:
feat: Add Files API support with upload, list, get and delete operations

- Implement complete Files API compatible with Gemini API format
- Support resumable file uploads with chunked transfer (tested with 15MB video)
- Create file management service with database tracking
- Add file domain models and API request/response objects
- Implement file routes with proper authentication
- Use fixed API key for Files API requests (due to Google API restrictions)
- Support file state management (PROCESSING, ACTIVE, FAILED)
- Add scheduled task for automatic expired file cleanup
- Integrate seamlessly with existing key management and load balancing

cxyfer committed:
Merge remote-tracking branch 'origin/main' into feature/upload-compatibility

cxyfer committed:
fix: Fix datetime.timezone AttributeError in file cleanup

- Change datetime.timezone.utc to timezone.utc in services.py
- Resolves error: 'type object datetime.datetime has no attribute timezone'

jqlts1 commented:
这个文件 api，和 key 是对应的吗？比如我上传时候的 key，和在使用的时候获取到的 key 是一样的吗？

cxyfer commented:
> 这个文件 api，和 key 是对应的吗？比如我上传时候的 key，和在使用的时候获取到的 key 是一样的吗？

是的，上传和使用时的 key 是对应的。

- 上传文件时，使用的 API key 会保存在数据库中
- 之后使用文件时，系统会自动取回相同的 API key
- 文件按 `user_token` 隔离，确保用户只能访问自己的文件

简单来说：哪个 key 上传，就用哪个 key 访问。

snailyp commented on app/service/files/files_service.py:
```diff
@@ -0,0 +1,501 @@
+        try:
+            async with AsyncClient() as client:
+                response = await client.get(
+                    f"https://generativelanguage.googleapis.com/v1beta/{file_name}",
```
可替换settings.BASE_URL
```suggestion
                    f"https://generativelanguage.googleapis.com/v1beta/{file_name}",
```

snailyp commented on app/service/files/files_service.py:
```diff
@@ -0,0 +1,501 @@
+                    
+                    async with AsyncClient() as client:
+                        await client.delete(
+                            f"https://generativelanguage.googleapis.com/v1beta/{file_name}",
```
这里可以改成settings.BASE_URL，更灵活一些
```suggestion
                            f"https://generativelanguage.googleapis.com/v1beta/{file_name}",
```

snailyp commented on app/router/files_routes.py:
```diff
@@ -0,0 +1,296 @@
+from app.domain.file_models import (
+    FileMetadata, 
+    ListFilesResponse, 
+    CreateFileRequest,
```
清理无用的包
```suggestion
    CreateFileRequest,
```

cxyfer committed:
refactor: Centralize API base URL and clean up

Replaces hardcoded Google API base URLs with `settings.BASE_URL` for improved configurability and maintainability across services.

Removes unused imports and variables from various modules to reduce code bloat and enhance readability.

fengwk commented:
我进行 upload 时候出现了网络错误
```
2025-10-06 23:23:53,062 | ERROR    | [files_service.py:179]         | Failed to initialize upload:
2025-10-06 23:23:53,062 | ERROR    | [files_routes.py:89]           | Upload initialization failed: Internal error:
```

我不太熟悉 python 但从变更代码看应该是没有尊重代理，变更里是这样构建 Client
```
async with AsyncClient() as client:
```

我对照了其它地方的构建方式是这样
```
async with httpx.AsyncClient(timeout=timeout, proxy=proxy) as client:
```

---

临时的解决方案是添加代理环境变量，我测试生效了，看起来 httpx 尊重了这些
```
export http_proxy=proxy_url
export https_proxy=proxy_url
```

暂时可以先把这些环境变量添加到 docker compose 里

fengwk commented:
<img width="1349" height="1850" alt="image" src="https://github.com/user-attachments/assets/a51f4ecb-b199-4d81-ab38-bf10d179dde7" />

如果在一轮对话中传递 2 张图就会出现问题，因为上传第二张图和第一张图负载均衡到的 apikey 不同，文件上传时也不会携带之前的图片因此无法定位到原 apikey，看起来在 gemini-balance 层没法解决这个问题？
